metadata-ingestion/src/datahub/ingestion/source/redshift/lineage.py

```diff
@@ -753,11 +712,13 @@ def _add_permanent_datasets_recursively(
                 db, schema, table = split_qualified_table_name(lineage_dataset.urn)
 
                 # Check if table found is again a temp table
-                if not RedshiftDataDictionary.is_empty(
-                    conn=connection,
-                    query=RedshiftQuery.is_temp_table_query(table=table),
-                ):
-                    transitive_temp_tables.append(table)
+                repeated_temp_table: List[TempTableRow] = self.find_temp_tables(
+                    connection=connection,
```

Comment:
Can you remove `connection` from here and pass in `temp_table_rows`? Connection here is quite confusing as we don't want to query the db again.

Reply:
done